pkg/verification/verification_service.go

```diff
@@ -0,0 +1,82 @@
+package verification
+
+import (
+	"crypto/rand"
+	"fmt"
+
+	"github.com/kevinburke/twilio-go"
+
+	"github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+)
+
+const (
+	codeCharset = "0123456789"
+	codeLength  = 6
+)
+
+// ServiceConfiguration represents the config used for the verification service.
+type ServiceConfiguration interface {
+	GetTwilioAccountSID() string
+	GetTwilioAuthToken() string
+	GetTwilioFromNumber() string
+	GetVerificationMessageTemplate() string
+}
+
+// Service represents the verification service for controllers.
+type Service interface {
+	SendVerification(signup *v1alpha1.UserSignup) error
+}
+
+// ServiceImpl represents the implementation of the verification service.
+type ServiceImpl struct {
+	config ServiceConfiguration
+}
+
+// NewVerificationService creates a service object for performing user verification
+func NewVerificationService(cfg ServiceConfiguration) (Service, error) {
```

Comment:
you don't need to return `error`, do you?

Reply:
Good point, fixed.